File: app/src/main/java/com/stevenfrew/beatprompter/AllSongsFilter.java
package com.stevenfrew.beatprompter;

import java.util.List;

class AllSongsFilter extends SongFilter
{
    AllSongsFilter(String name,List<SongFile> songs)
    {
        super(name,songs,true);
    }
    public boolean equals(Object o) {
        return o == null || o instanceof AllSongsFilter;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/AllSongsFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.BeatPrompter
import com.stevenfrew.beatprompter.R
import com.stevenfrew.beatprompter.cache.SongFile

class AllSongsFilter(songs: List<SongFile>) : SongFilter(
	BeatPrompter.appResources.getString(R.string.no_tag_selected),
	songs, true
) {
	override fun equals(other: Any?): Boolean = other == null || other is AllSongsFilter

	override fun hashCode(): Int = javaClass.hashCode()
}